### PR TITLE
HBASE-22338 LICENSE file only contains Apache 2.0

### PR DESCRIPTION
**Pull request by taklwu**
- fix NOTICE.aggregate generation

please see the other discussion on #41 , and I think #41 should be closed and continue the discussion here.

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   1m 48s |  master passed  |
| +1 :green_heart: |  compile  |   1m  1s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m  6s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 59s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  1s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  1s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m 55s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   9m 45s |  root in the patch passed.  |
|  |   |  20m 49s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-93/1/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-connectors/pull/93 |
| Optional Tests | dupname javac javadoc unit xml compile |
| uname | Linux 9084a66a50d4 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 GNU/Linux |
| Build tool | hb_maven |
| Personality | dev-support/jenkins/hbase-personality.sh |
| git revision | master / 2bfc5f1 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-93/1/testReport/ |
| Max. process+thread count | 916 (vs. ulimit of 12500) |
| modules | C: hbase-connectors-assembly . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-93/1/console |
| versions | git=2.20.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: pom.xml [hunk 1]
```diff
@@ -128,6 +128,15 @@
     <buildDate>${maven.build.timestamp}</buildDate>
     <compileSource>1.8</compileSource>
     <java.min.version>${compileSource}</java.min.version>
+    <!-- override on command line to have generated LICENSE files include
+         diagnostic info for verifying notice requirements -->
+    <build.year>2022</build.year>
```
Comment (busbey) [inline pom.xml:133]: we set this already via the `buildnumber-maven-plugin`. is that not working?

**Review by busbey — COMMENTED** [review]
Do you have examples of the generated LICENSE/NOTIC files after this change?

I'll build locally in the mean time to get them.

File: pom.xml (hunk 1 above)
Comment (taklwu) [reply pom.xml:133]: I don't know why it's not working yet, but here is the output https://issues.apache.org/jira/secure/attachment/13040127/NOTICE.aggregate-no-build-year

```
Apache HBase Connectors - Assembly
Copyright 2018-${copyright-end-year} The Apache Software Foundation
```


let me continue to check it.

Comment (taklwu) [reply pom.xml]: I can't the reason -.- although I tried to use `-Dbuild.year=2022` and it worked.... I will revert the change for this build.year first. let me know if you find any thing but my goal is to have the dependency included included in the `NOTICE` file correctly.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 41s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 25s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m  0s |  master passed  |
| +1 :green_heart: |  compile  |   1m 18s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 19s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  9s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m  6s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 20s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 20s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   1m 23s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   9m 33s |  root in the patch passed.  |
|  |   |  21m 31s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-93/2/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-connectors/pull/93 |
| Optional Tests | dupname javac javadoc unit xml compile |
| uname | Linux 645239bb5a3d 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 GNU/Linux |
| Build tool | hb_maven |
| Personality | dev-support/jenkins/hbase-personality.sh |
| git revision | master / 2bfc5f1 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-93/2/testReport/ |
| Max. process+thread count | 912 (vs. ulimit of 12500) |
| modules | C: hbase-connectors-assembly . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-93/2/console |
| versions | git=2.20.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 38s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 25s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   1m 42s |  master passed  |
| +1 :green_heart: |  compile  |   1m  0s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 52s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  5s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 48s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 53s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 53s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m 49s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   8m 35s |  root in the patch passed.  |
|  |   |  17m 43s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-93/3/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-connectors/pull/93 |
| Optional Tests | dupname javac javadoc unit xml compile |
| uname | Linux 49fc9edceba8 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 GNU/Linux |
| Build tool | hb_maven |
| Personality | dev-support/jenkins/hbase-personality.sh |
| git revision | master / 2bfc5f1 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-93/3/testReport/ |
| Max. process+thread count | 920 (vs. ulimit of 12500) |
| modules | C: hbase-connectors-assembly . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-93/3/console |
| versions | git=2.20.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: pom.xml (hunk 1 above)
Comment (taklwu) [reply pom.xml]: ping ?

**Comment by busbey** [comment]
Here's what I see when verifying things.

Starting with the branch for this PR:
```
% git clean -xdf
...
% mvn package verify
...
% cd hbase-connectors-assembly/target
% tar xzf hbase-connectors-1.0.1-SNAPSHOT-bin.tar.gz
% ls hbase-connectors-1.0.1-SNAPSHOT/lib | grep -v -E "^hbase-.*\.jar" | wc -l # confirm there are third party jars
      68
% cat hbase-connectors-1.0.1-SNAPSHOT/NOTICE.txt 
Apache HBase - Connectors
Copyright 2019 The Apache Software Foundation
% vim hbase-connectors-1.0.1-SNAPSHOT/LICENSE.txt  # it's just the ALv2 license text

**Review by busbey — CHANGES_REQUESTED** [review]
I don't think this currently solves the issue with the generated assembly tarball

**Comment by taklwu** [comment]
sorry, I'm learning toward this JIRA as well, what you mean here is that we need a NOTICE.txt/NOTICE after unzipping the tar ball. 

basically, I knew that the NOTICE file were generated at `target/NOTICE.aggregate` that was not correctly generated before this change. Let me come back on it and see how I can rename or move this `target/NOTICE.aggregate` to NOTICE. 

but for the LICENSE file, I may need another look.

**Comment by taklwu** [comment]
yup, you're right, I missed that file copy in the assembly tar, I checked it but please double check as well 

```
% pwd
/Users/taklwu/workspace/hbase-connectors/hbase-connectors-assembly/target/hbase-connectors-1.0.1-SNAPSHOT

% ls -l
total 224
-rw-r--r--   1 taklwu  staff  12119 Jan 22  2020 CHANGELOG.md
-rw-r--r--   1 taklwu  staff      0 Jan 22  2020 LEGAL
-rw-r--r--   1 taklwu  staff  67420 Jan 22  2020 LICENSE.txt
-rw-r--r--   1 taklwu  staff  19709 Jan 22  2020 NOTICE.txt
-rw-r--r--   1 taklwu  staff   1006 Jan 22  2020 README.md
-rw-r--r--   1 taklwu  staff   6845 Jan 22  2020 RELEASENOTES.md
drwxr-xr-x   5 taklwu  staff    160 Feb 24 14:28 bin
drwxr-xr-x   3 taklwu  staff     96 Jan 22  2020 conf
drwxr-xr-x  89 taklwu  staff   2848 Feb 24 14:28 lib

% diff NOTICE.txt ../NOTICE.aggregate # no difference 
```

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m  1s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   1m 23s |  master passed  |
| +1 :green_heart: |  compile  |   0m 52s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 51s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  6s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 48s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 52s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 52s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m 50s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   8m 33s |  root in the patch passed.  |
|  |   |  17m 34s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-93/4/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-connectors/pull/93 |
| Optional Tests | dupname javac javadoc unit xml compile |
| uname | Linux 5c6f28987365 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 GNU/Linux |
| Build tool | hb_maven |
| Personality | dev-support/jenkins/hbase-personality.sh |
| git revision | master / 2bfc5f1 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-93/4/testReport/ |
| Max. process+thread count | 920 (vs. ulimit of 12500) |
| modules | C: hbase-connectors-assembly . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-93/4/console |
| versions | git=2.20.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by busbey** [comment]
we're missing the part of the main project tooling that checks for problems with the generated license:

```
(base) sbusbey@seans-mbp hbase-connectors-1.0.1-SNAPSHOT % grep -B 2 "^ERROR:" LICENSE.txt
This product includes Streaming API for XML licensed under the GNU General Public Library.

ERROR: Please check ^^^^^^^^^^^^ this License for acceptability here:
```

stax-api is dual licensed GPL and CDDL. We'll also need to specify that we're picking CDDL.

The generated NOTICE is missing the build year:

```
(base) sbusbey@seans-mbp hbase-connectors-1.0.1-SNAPSHOT % head -n 3 NOTICE.txt 

Apache HBase Connectors - Assembly
Copyright 2018-${copyright-end-year} The Apache Software Foundation
```
It looks like this is because we need to activate the `buildnumber-maven-plugin` in the `hbase-connectors-assembly` module. we only define the version/configs in plugin management on the top level pom; we never actually tell the plugin to run.

**Comment by taklwu** [comment]
for 2/ about the copyright-end-year has been fixed, but for 1/ of LICENSE.txt, this package does not have a supplemental-models.xml and it's inherited from `hbase-resource-bundle`.

I'm wondered if you have any suggestion here, my current thought is to copy the supplemental-models.xml from HBase repo, but I'm checking with you at the same time before I find the solution.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 39s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 27s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   1m 38s |  master passed  |
| +1 :green_heart: |  compile  |   1m  1s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m  7s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  6s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m  2s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  9s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  9s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  5s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m 58s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   8m 58s |  root in the patch passed.  |
|  |   |  19m 15s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-93/5/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-connectors/pull/93 |
| Optional Tests | dupname javac javadoc unit xml compile |
| uname | Linux 2f401baf0e5e 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 GNU/Linux |
| Build tool | hb_maven |
| Personality | dev-support/jenkins/hbase-personality.sh |
| git revision | master / 2bfc5f1 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-93/5/testReport/ |
| Max. process+thread count | 916 (vs. ulimit of 12500) |
| modules | C: hbase-connectors-assembly . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-93/5/console |
| versions | git=2.20.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by taklwu** [comment]
@busbey I was out last weekend, sorry for the delay, but can you help to check again, I should have fixed what you commented.

```
taklwu@MacBook-Pro hbase-connectors-1.0.1-SNAPSHOT % grep -B 2 "^ERROR:" LICENSE.txt
taklwu@MacBook-Pro hbase-connectors-1.0.1-SNAPSHOT % head -n 3 NOTICE.txt

Apache HBase Connectors - Assembly
Copyright 2018-2022 The Apache Software Foundation
```

**Review by busbey — CHANGES_REQUESTED** [review]
The generated files look correct. But we still need to make sure the build will fail if we add some new dependency that is under a category-x license.

this enforcer check is what does it in the main project: https://github.com/apache/hbase/blob/rel/3.0.0-alpha-2/pom.xml#L1316

**Comment by taklwu** [comment]
add the goal of `check-aggregate-license`, and if we remove the local `supplemental-models.xml` , failed as expected. 

```
[INFO] --- maven-enforcer-plugin:3.0.0:enforce (check-aggregate-license) @ hbase-connectors-assembly ---
[WARNING] Rule 0: org.apache.maven.plugins.enforcer.EvaluateBeanshell failed with message:
License errors detected, for more detail find ERROR in
                    /Users/taklwu/workspace/hbase-connectors/hbase-connectors-assembly/target/maven-shared-archive-resources/META-INF/LICENSE
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.543 s
[INFO] Finished at: 2022-03-01T18:11:45-08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0:enforce (check-aggregate-license) on project hbase-connectors-assembly: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

**Comment by taklwu** [comment]
checking again, @busbey  do you see if there is any other problem ?

**Review by busbey — APPROVED** [review]

**Pull request merged**